Mathlib/Analysis/NormedSpace/ProdLp.lean

```diff
@@ -0,0 +1,777 @@
+/-
+Copyright (c) 2023 Moritz Doll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Doll, Sébastien Gouëzel, Jireh Loreaux
+-/
+
+import Mathlib.Analysis.MeanInequalities
+import Mathlib.Analysis.NormedSpace.WithLp
+
+/-!
+# `L^p` distance on products of two metric spaces
+Given two metric spaces, one can put the max distance on their product, but there is also
+a whole family of natural distances, indexed by a parameter `p : ℝ≥0∞`, that also induce
+the product topology. We define them in this file. For `0 < p < ∞`, the distance on `α × β`
+is given by
+$$
+d(x, y) = \left(d(x_1, y_1)^p + d(x_2, y_2)^p\right)^{1/p}.
+$$
+For `p = ∞` the distance is the supremum of the distances and `p = 0` the distance is the
+cardinality of the elements that are not equal.
+
+We give instances of this construction for emetric spaces, metric spaces, normed groups and normed
+spaces.
+
+To avoid conflicting instances, all these are defined on a copy of the original Prod-type, named
+`WithLp p (α × β)`. The assumption `[Fact (1 ≤ p)]` is required for the metric and normed space
+instances.
+
+We ensure that the topology, bornology and uniform structure on `WithLp p (α × β)` are (defeq to)
+the product topology, product bornology and product uniformity, to be able to use freely continuity
+statements for the coordinate functions, for instance.
+
+# Implementation notes
+
+This files is a straight-forward adaption of `Mathlib.Analysis.NormedSpace.PiLp`.
+
+-/
+
+local macro_rules | `($x ^ $y)   => `(HPow.hPow $x $y) -- Porting note: See issue #2220
```

Comment:
Can you use the same comment we use elsewhere, that clarifies which repo this issue number refers to (it's not #2220)